src/main/kotlin/com/infusionvlc/somniumserver/dreams/usecases/CreateDream.kt

```diff
@@ -28,7 +30,12 @@ class CreateDream(
       .flatMap { dream ->
         findUserById.execute(userId)
           .toEither { DreamCreationErrors.CreatorNotFound(userId) }
-          .map { user -> dao.save(dream.toEntity(user)) }
+          .map { user ->
+            dreamRequest.tags.forEach { tag ->
+              createTag.execute(tag)
+            }
```

Comment:
What happens if creating one tag fails?

Reply:
🔥⚠️throw an error ⚠️🔥

Throwing is an effect and we don't want effects because we are cool functional guys 😎.

Maybe we should `map` over dreamRequests.tags instead of `forEach` because that returns Unit and everything which does not return anything is an effect. But that leaves us with a `List<Either<???Error, Tag>>` and that can be quite complex to handle. (Not really but we may start seeing big names like `Traversable` or `Monad Transformers`). I'll take care of this when I can.

Reply:
With an error I wanted to say a response error not throwing a normal error.

I tried to do it with the `map` but I encountered the problem that you say so I thinked that better to upload this and improve it.